pennylane/vqe/vqe.py

```diff
@@ -0,0 +1,217 @@
+# Copyright 2019 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This submodule contains functionality for running Variational Quantum Eigensolver (VQE)
+computations using PennyLane.
+"""
+# pylint: disable=too-many-arguments, too-few-public-methods
+import numpy as np
+import pennylane as qml
+from pennylane.operation import Observable, Tensor
+
+
+OBS_MAP = {
+    "PauliX": "X",
+    "PauliY": "Y",
+    "PauliZ": "Z",
+    "Hadamard": "H",
+    "Identity": "I"
+}
+
+
+class Hamiltonian:
+    r"""Lightweight class for representing Hamiltonians for Variational Quantum
+    Eigensolver problems.
+
+    Hamiltonians can be expressed as linear combinations of observables, e.g.,
+    :math:`\sum_{k=0}^{N-1} c_k O_k`.
+
+    This class keeps track of the terms (coefficients and observables) separately.
+
+    Args:
+        coeffs (Iterable[float]): coefficients of the Hamiltonian expression
+        observables (Iterable[Observable]): observables in the Hamiltonian expression
+
+    .. seealso:: :class:`~.VQECost`, :func:`~.generate_hamiltonian`
```

Comment:
Will the second reference resolve if QChem package is not installed??

Reply:
No, it will produce an `ImportError`. I can modify this to produce a more useful error message.

Reply:
I've looked into this, and it's not entirely trivial. We need something that raises the correct error message for an arbitrary number of 'attribute chains', i.e., `qml.qchem` and `qml.qchem.generate_hamiltonian` should raise the same error message.

This is doable (we did something similar when we deprecated `qml.expval.something`), but probably large enough to be out of scope for this PR . I suggest implementing this (as well as the corresponding tests) in a new PR.

Reply:
Oh wait, I misunderstood. Do you mean will the link resolve in the docs if qchem is not installed? It will not, but it will always resolve on readthedocs, as qchem is installed there. (And we should write the docs with the online version in mind, since that is the one most users will access.)